lib/class.NGIpublications.php

```diff
@@ -408,28 +553,52 @@ public function formatPublication($publication,$details=FALSE) {
 			$tools=new htmlElement('div');
 			$tools->set('class','large-2 columns');
 
+			//Content
 			$title=new htmlElement('h5');
-			$title->set('text',$publication_status.'<span class="label">'.$publication['data']['score'].'</span> '.html_entity_decode($publication['data']['title']));
+			$title->set('text',$publication_status.'<span class="label">'.$publication['data']['score'].'</span> '.html_entity_decode($publication['data']['title']).' (<a href="https://www.ncbi.nlm.nih.gov/pubmed/'.$publication['data']['pmid'].'">Pubmed</a>)');
 
 			$ref=new htmlElement('p');
-			if($details) {
-				$ref->set('text',implode(', ', $publication['authors']).'<br>'.date('Y',strtotime($publication['data']['pubdate'])).', '.$publication['data']['journal'].', '.$reference);
-			} else {
-				$ref->set('text',$publication['authors'][0].' et. al. '.date('Y',strtotime($publication['data']['pubdate'])).', '.$publication['data']['journal'].', '.$reference);
-			}
+			$ref->set('text',$publication['authors'][0].' et. al. '.date('Y',strtotime($publication['data']['pubdate'])).', '.$publication['data']['journal'].', '.$reference);
+			
+			$authors=new htmlElement('p');
+			$authors->set('text',implode(', ', $publication['authors']).'<br>');
 
-			$researchers=new htmlElement('p');
-			$researchers->set('text','Matched authors: '.$researcher_string.'<br>Matched keywords: '.$keyword_string);
+			$abstract=new htmlElement('p');
+			$abstract->set('text',$publication['data']['abstract']);
 
-			$tools_read=new htmlElement('a');
-			$tools_read->set('href','publications.php?id='.$publication['data']['id']);
-			$tools_read->set('class','tiny button expanded');
-			$tools_read->set('text','Read');
+			$researchers=new htmlElement('p');
+			$researchers->set('text','Matched authors: '.$researcher_string.'<br>Matched keywords in abstract: '.$keyword_string);
+			
+			// Fulltext keyword matches
+			$fulltext = new htmlElement('div');
+			$matches = json_decode($publication['fulltext']['text']);
+			$amatches = [];
+			foreach($matches as $match) {
+				$amatches[$match[0]][] = $match[1];
+			}
+			$bmatches = array_map(function($ar){ return implode("...<br/><br/>",$ar); }, $amatches);
+			if($bmatches) {
+				$fulltext_keywords=new zurbAccordion(TRUE,TRUE);
+				foreach($bmatches as $key => $match){
+					$mtext = preg_replace('/'.$key.'/i', '<mark>${0}</mark>', $match);
+					$fulltext_keywords->addAccordion($key,$mtext);
+				}
+				$fulltext->set('text','<strong>Managed to retrieve fulltext of this paper, see matched keywords in list below!</strong>'.$fulltext_keywords->render());
+			} else {
+				$fulltext->set('text','<strong>No matches in fulltext</strong>');
```

Comment:
Thanks @ormestad for fixing this. :+1: I just quickly clobbered this part together.